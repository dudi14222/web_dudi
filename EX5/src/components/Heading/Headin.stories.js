import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import {
  withKnobs,
  boolean, number,text
} from '@kadira/storybook-addon-knobs';

import Heading from './';


const stories = storiesOf('Heading', module);
stories.addDecorator(withKnobs);

stories.addWithInfo('with some props', '', () => {
  const story = <Heading
        className={text('class', 'section')}
        size={number('size', 2)}


    >
        <div>children</div>
  </Heading>;
  return story;
});
