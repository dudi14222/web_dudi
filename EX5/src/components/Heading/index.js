import React from 'react';
import classNames from 'classnames';
import './headings.css';

const Heading = ({ size, children, className, ...props }) => 
    React.createElement(
        `h${size}`,{ 
            className: classNames([
                className, `heading heading-${size}`
            ]), ...props 
        } , children
    )
export default Heading;    