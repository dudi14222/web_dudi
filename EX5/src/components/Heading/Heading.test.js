import Heading from '../Heading';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('Heading snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Heading />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('Heading shallow tests', () => {
    it('should render children when passed in', () => {
        const wrapper = shallow((
            <Heading>
                <div className="unique" />
            </Heading>
        ));    
        expect(wrapper.contains(<div className="unique" />)).toBe(true);
    });
})



