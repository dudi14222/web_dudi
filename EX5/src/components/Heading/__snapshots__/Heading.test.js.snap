// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading snapshot tests test snapshot 1`] = `
<hundefined
  className="heading heading-undefined"
/>
`;
